src/Cryptol/Eval/SBV.hs

```diff
@@ -619,11 +619,13 @@ updateFrontSym_word (Nat n) eltTy bv wv val =
       | WordVal bw <- bv ->
         WordVal <$>
           do b <- fromVBit <$> val
-             let sz = SBV.intSizeOf bw
-             let q = SBV.svSymbolicMerge (SBV.kindOf bw) True b bw (literalSWord sz 0)
-             let msk = SBV.svShiftRight (literalSWord sz (bit (sz-1))) idx
-             let bw' = SBV.svXOr bw (SBV.svAnd q msk)
-             return $! bw'
+             let sz   = SBV.intSizeOf bw
+             let z    = literalSWord sz 0
+             let znot = SBV.svNot z
+             let q    = SBV.svSymbolicMerge (SBV.kindOf bw) True b znot z
+             let msk  = SBV.svShiftRight (literalSWord sz (bit (sz-1))) idx
+             let bw'  = SBV.svAnd bw (SBV.svNot msk)
+             return $! SBV.svXOr bw' (SBV.svAnd q msk)
```

Comment:
Was this a bug that could make Cryptol produce wrong results in practice? If so, we ought to have a regression test for it (probably also a github ticket).

Reply:
This case didn't exist before, and would just fall back to an unpacked representation.

Either way, you're right, we should add a test case.

Reply:
Test added. And, indeed, the behavior prior to 0ce797f was incorrect.